### DISCUSSION TITLE
Gallery Problem - Incorrect Width and Height in EXIF Data Affecting PhotoSwipe Display

### DISCUSSION THREAD

**Discussion by Muensteraner** [GetPublii/Publii#1694] [Bug Report] [answered]
### Operating system

Win 7

### Publii version

0.46.1

### Issue type

Application

### Bug description

I have a problem with portrait and lengthwise pictures in the gallery. The portrait images are always shown stretched across the entire width. What is the problem and how can you solve the problem if you want to use different portrait and landscape photos in the gallery and then display them enlarged.

Sample side: http://m-learning.info/2.html

**Comment by bjazmoore** [comment]
What theme are you using?

**Comment by Muensteraner** [comment]
Simple... i tried all versions

**Comment by bobmitro** [comment, marked as answer]
It's not a theme issue or, to be more specific, the PhotoSwipe script. The problem is with the image's EXIF data, precisely the width and height recorded in the EXIF info. 

For example, look at this:

<img width="953" alt="screenshot 2024-09-16 at 08 27 20" src="https://github.com/user-attachments/assets/8725b486-fc78-4111-b7a2-e7471995776e">

The EXIF data of your original image shows the wrong width and height, it should be the other way around.

Now, this is how it should look:

<img width="998" alt="screenshot 2024-09-16 at 08 28 08" src="https://github.com/user-attachments/assets/225b98b4-974f-45dd-8204-bcc63d453e98">

Below, download the image with the corrected width/height EXIF data and see that it will now render well in the gallery.

![image](https://github.com/user-attachments/assets/e0b527d3-89bf-4c5e-970c-0775e4b848c1)

**Comment by myelin** [comment]
Can you double check this? I downloaded the problematic image (http://m-learning.info/media/posts/2/gallery/20240905_115629.jpg) and ran `identify` from ImageMagick on it, and its info looks correct:

```
Tulin:~/Downloads phil$ identify 20240905_115629.jpg
20240905_115629.jpg JPEG 1365x768 1365x768+0+0 8-bit sRGB 278783B 0.000u 0:00.001
Tulin:~/Downloads phil$ identify -verbose 20240905_115629.jpg | grep Orientation
  Orientation: RightTop
```

The image data has a width of 1365 pixels and height of 768 pixels, and pixel (0,0) is the top-right corner, i.e. it needs to be rotated 90 degrees counter-clockwise for display, and the displayed image will have width=768 height=1365.

This is the same as what I'm seeing for portrait-type images exported as JPEG from Apple Photos, so photos with metadata like this are probably pretty common.

Simon Willison has a good explanation of what's going on here, and his own experiences with Photoswipe requiring orientation-aware width and height to display images correctly: https://til.simonwillison.net/exif/orientation-and-location

I've opened a pull request (https://github.com/GetPublii/Publii/pull/2039) to wrap the sizeofOf() function from the image-size module and swap width/height as necessary.
